### PR TITLE
Clean up Api modules.

### PR DESCRIPTION
**Pull request by florian-kuebler**
So far, Api consists of multiple modules inside the same directory.
This change splits this into distinct directories, renamed ApiBase
to ApiUtils and made the separation more clear.

Test: Compile and profile OrbitTest(with introspection).
Bug: http://b/196031933

### CODE REVIEW

**Review by beckerhe — COMMENTED** [review]
Structure looks good but I'm not sure about the include paths. We should strictly avoid `/../` components in include paths. But to me it also looks like they aren't needed.

File: src/Api/LockFreeApiEventProducer.h [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 #include <variant>
 
-#include "Api/Event.h"
+#include "../ApiUtils/include/ApiUtils/Event.h"
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/Api/LockFreeApiEventProducer.h]: Done.

**Comment by beckerhe** [comment]
> > Structure looks good but I'm not sure about the include paths. We should strictly avoid `/../` components in include paths. But to me it also looks like they aren't needed.
> 
> Oh, that seem to was the automatic refactoring. I'll take care of it! Thanks for the review so far!

Ah yeah. This was my suspicion. :smile:

File: src/Api/Orbit.cpp [hunk 2]
```diff
@@ -3,11 +3,11 @@
 // found in the LICENSE file.
 
 #define ORBIT_API_INTERNAL_IMPL
-#include "Api/Orbit.h"
+#include "../ApiInterface/include/ApiInterface/Orbit.h"
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/Api/Orbit.cpp]: Done.

File: src/Api/Orbit.cpp [hunk 3]
```diff
@@ -3,11 +3,11 @@
...
 
 #include <absl/base/casts.h>
 
-#include "Api/Event.h"
+#include "../ApiUtils/include/ApiUtils/Event.h"
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/Api/Orbit.cpp]: Done.

File: src/ApiUtils/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,28 @@
+add_library(ApiUtils STATIC)
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/ApiUtils/CMakeLists.txt:5]: Thanks. Done.

File: src/ApiUtils/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+target_sources(ApiUtils PUBLIC
+        include/ApiUtils/Event.h
+        include/ApiUtils/EncodedEvent.h
+        include/ApiUtils/EncodedString.h)
+target_sources(ApiUtils PRIVATE
+        EncodedString.cpp)
+target_link_libraries(ApiUtils PUBLIC
+        GrpcProtos
+        OrbitBase)
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/ApiUtils/CMakeLists.txt]: Good catch. Thanks!

File: src/ApiUtils/EncodedEventTest.cpp [hunk 6]
```diff
@@ -8,7 +8,7 @@
 
 #include <string>
 
-#include "Api/EncodedEvent.h"
+#include "include/ApiUtils/EncodedEvent.h"
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/ApiUtils/EncodedEventTest.cpp]: Done.

File: src/ApiUtils/EncodedString.cpp [hunk 7]
```diff
@@ -2,14 +2,14 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#include "include/Api/EncodedString.h"
+#include "include/ApiUtils/EncodedString.h"
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/ApiUtils/EncodedString.cpp]: Done.

File: src/ApiUtils/EncodedString.cpp [hunk 8]
```diff
@@ -2,14 +2,14 @@
...
 
 #include <absl/base/casts.h>
 
 #include <cstring>
 
 #include "capture.pb.h"
-#include "include/Api/Event.h"
+#include "include/ApiUtils/Event.h"
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/ApiUtils/EncodedString.cpp]: Done.

File: src/ApiUtils/EncodedStringTest.cpp [hunk 9]
```diff
@@ -7,7 +7,7 @@
 #include <gtest/gtest.h>
 
 #include "capture.pb.h"
-#include "include/Api/EncodedString.h"
+#include "include/ApiUtils/EncodedString.h"
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/ApiUtils/EncodedStringTest.cpp]: Done.

File: src/ApiUtils/include/ApiUtils/EncodedEvent.h [hunk 10]
```diff
@@ -2,14 +2,14 @@
...
-#define ORBIT_API_ENCODED_EVENT_H_
+#ifndef ORBIT_API_UTILS_ENCODED_EVENT_H_
+#define ORBIT_API_UTILS_ENCODED_EVENT_H_
 
 #include <stdint.h>
 
 #include <cstring>
 
-#include "Orbit.h"
+#include "../../../ApiInterface/include/ApiInterface/Orbit.h"
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/ApiUtils/include/ApiUtils/EncodedEvent.h]: Done.

File: src/ApiUtils/include/ApiUtils/Event.h [hunk 11]
```diff
@@ -2,20 +2,20 @@
...
-#ifndef ORBIT_API_EVENT_H_
-#define ORBIT_API_EVENT_H_
+#ifndef ORBIT_API_UTILS_EVENT_H_
+#define ORBIT_API_UTILS_EVENT_H_
 
 #include <cstdint>
 #include <variant>
 #include <vector>
 
+#include "../../../ApiInterface/include/ApiInterface/Orbit.h"
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/ApiUtils/include/ApiUtils/Event.h]: Done.

File: src/CaptureClient/ApiEventProcessor.cpp [hunk 12]
```diff
@@ -4,7 +4,7 @@
 
 #include "CaptureClient/ApiEventProcessor.h"
 
-#include "Api/EncodedString.h"
+#include "../ApiUtils/include/ApiUtils/EncodedString.h"
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/CaptureClient/ApiEventProcessor.cpp]: Done.

File: src/CaptureClient/ApiEventProcessorTest.cpp [hunk 13]
```diff
@@ -9,7 +9,7 @@
 #include <algorithm>
 #include <vector>
 
-#include "Api/EncodedString.h"
+#include "../ApiUtils/include/ApiUtils/EncodedString.h"
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/CaptureClient/ApiEventProcessorTest.cpp]: Done.

File: src/CaptureClient/include/CaptureClient/ApiEventProcessor.h [hunk 14]
```diff
@@ -7,7 +7,7 @@
 
 #include <absl/container/flat_hash_map.h>
 
-#include "Api/EncodedEvent.h"
+#include "../../../ApiUtils/include/ApiUtils/EncodedEvent.h"
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/CaptureClient/include/CaptureClient/ApiEventProcessor.h]: Done.

File: src/Introspection/include/Introspection/Introspection.h [hunk 15]
```diff
@@ -10,8 +10,8 @@
 #include <functional>
 #include <memory>
 
-#include "Api/EncodedEvent.h"
-#include "Api/Orbit.h"
+#include "../../../ApiInterface/include/ApiInterface/Orbit.h"
+#include "../../../ApiUtils/include/ApiUtils/EncodedEvent.h"
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/Introspection/include/Introspection/Introspection.h]: Done.

File: src/OrbitGl/CGroupAndProcessMemoryTrack.cpp [hunk 16]
```diff
@@ -7,7 +7,7 @@
 #include <absl/strings/str_format.h>
 #include <absl/strings/substitute.h>
 
-#include "Api/EncodedEvent.h"
+#include "../ApiUtils/include/ApiUtils/EncodedEvent.h"
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/OrbitGl/CGroupAndProcessMemoryTrack.cpp]: Done.

**Review by florian-kuebler — COMMENTED** [review]
Thanks the comment. I addressed them.

**Review by beckerhe — APPROVED** [review]
LGTM

**Review by pierricgimmig — APPROVED** [review]
Please modify build.sh with the new location of Orbit.h, otherwise, lgtm.

**Pull request merged**